app/models/user.rb

```diff
@@ -19,6 +19,21 @@ def capitalize_name
     self.last_name = last_name.capitalize
   end
 
+  def self.from_omniauth(auth)
+    uid = auth.uid
+    email = auth.info.email
+    user = User.find_by(email:)
+
+    if user
+      user.update(uid: nil) unless user.uid == uid
+      return user, uid
+    end
+
+    password = Devise.friendly_token[0, 20]
+    first_name, *last_name = auth.info.name.split
+    User.create!(email:, password:, first_name:, last_name: last_name.join(" "), provider: auth.provider, uid:)
+  end
```

Comment:
@cipriancrisan what this method does is when I user tries to log in with facebook it calls this method from the Users::OmniauthCallbacksController.
If a user already has this facebook uid then we return it, if not then we create a new user because even if the user logs in with facebook we still wanna save it to our database